examples/stable_diffusion_v2/inference/libs/sd_models.py

```diff
@@ -0,0 +1,228 @@
+        return image_latents.astype(ms.float16)
+
+    @ms.jit
+    def vae_decode(self, x):
+        y = self.vae.decode(x / self.scale_factor)
+        y = ops.clip_by_value((y + 1.0) / 2.0, clip_value_min=0.0, clip_value_max=1.0)
+        return y
+
+    @ms.jit
+    def latents_add_noise(self, image_latents, noise, ts):
+        latents = self.scheduler.add_noise(image_latents, noise, self.alphas_cumprod[ts])
+        return latents
+
+    @ms.jit
+    def predict_noise(self, x, t_continuous, condition, unconditional_condition):
+        """
+        The noise predicition model function that is used for DPM-Solver.
+        """
+        t_continuous = ops.tile(t_continuous.reshape(1), (x.shape[0],))
+        if self.guidance_scale == 1.0:
+            return self.unet(x, t_continuous, c_crossattn=condition)
+        x_in = ops.concat([x] * 2, axis=0)
+        t_in = ops.concat([t_continuous] * 2, axis=0)
+        c_in = ops.concat([unconditional_condition, condition], axis=0)
+        noise_pred = self.unet(x_in, t_in, c_crossattn=c_in)
+        noise_pred_uncond, noise_pred_text = ops.split(noise_pred, split_size_or_sections=noise_pred.shape[0] // 2)
+        noise_pred = noise_pred_uncond + self.guidance_scale * (noise_pred_text - noise_pred_uncond)
+        if self.guidance_rescale > 0:
+            noise_pred = self.rescale_noise_cfg(noise_pred, noise_pred_text)
+        return noise_pred
+
+    def rescale_noise_cfg(self, noise_pred, noise_pred_text):
+        """
+        Rescale `noise_pred` according to `guidance_rescale`. Based on findings of [Common Diffusion Noise Schedules and
+        Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf). See Section 3.4
+        """
+        std_text = ops.std(noise_pred_text, axis=tuple(range(1, len(noise_pred_text.shape))), keepdims=True)
+        std_cfg = ops.std(noise_pred, axis=tuple(range(1, len(noise_pred.shape))), keepdims=True)
+        # rescale the results from guidance (fixes overexposure)
+        noise_pred_rescaled = noise_pred * (std_text / std_cfg)
+        # mix with the original results from guidance by factor guidance_rescale to avoid "plain looking" images
+        noise_pred = self.guidance_rescale * noise_pred_rescaled + (1 - self.guidance_rescale) * noise_pred
+        return noise_pred
+
+    def __call__(self, inputs):
+        pass
+
+
+class SD_Text2Img(SD_Infer):
+    def __init__(
+        self,
+        text_encoder,
+        unet,
+        vae,
+        scheduler,
+        scale_factor=1.0,
+        guidance_scale=7.5,
+        guidance_rescale=0.0,
+        num_inference_steps=50,
+    ):
+        super(SD_Text2Img, self).__init__(
+            text_encoder,
+            unet,
+            vae,
+            scheduler,
+            scale_factor=scale_factor,
+            guidance_scale=guidance_scale,
+            guidance_rescale=guidance_rescale,
+            num_inference_steps=num_inference_steps,
+        )
+
+    def __call__(self, inputs):
```

Comment:
`SD_Text2Img`, `SD_Img2Img`, and `SD_Inpaint`'s `__call__` differ in the initial latents only. Can also combine them as I proposed in https://github.com/CaitinZhao/mindone/pull/1.

Reply:
has refactored